Products/contentmigration/common.py

```diff
@@ -93,44 +103,69 @@ def unrestricted_rename(self, id, new_id):
         * no verify object check from PortalFolder so it's allowed to rename
           even unallowed portal types inside a folder
     """
-    try: self._checkId(new_id)
-    except: raise CopyError(MessageDialog(
-                  title='Invalid Id',
-                  message=sys.exc_info()[1],
-                  action ='manage_main'))
-    ob=self._getOb(id)
-    #!#if ob.wl_isLocked():
-    #!#    raise(ResourceLockedError, 'Object "%s" is locked via WebDAV' % ob.getId())
+    try:
+        self._checkId(new_id)
+    except Exception:
+        raise CopySupport.CopyError(Dialogs.MessageDialog(
+            title='Invalid Id',
+            message=sys.exc_info()[1],
+            action='manage_main'))
+
+    ob = self._getOb(id)
+
     if not ob.cb_isMoveable():
-        raise CopyError('Not supported {}'.format(escape(id)))
-    #!#self._verifyObjectPaste(ob)
-    #!#CopyContainer._verifyObjectPaste(self, ob)
-    try:    ob._notifyOfCopyTo(self, op=1)
-    except: raise(CopyError, MessageDialog(
-                  title='Rename Error',
-                  message=sys.exc_info()[1],
-                  action ='manage_main'))
-    self._delObject(id)
+        raise CopySupport.CopyError(
+            CopySupport.eNotSupported % cgi.escape(id))
+
+    try:
+        ob._notifyOfCopyTo(self, op=1)
+    except POSException.ConflictError:
+        raise
+    except Exception:
+        raise CopySupport.CopyError(Dialogs.MessageDialog(
+            title="Rename Error",
+            message=sys.exc_info()[1],
+            action='manage_main'))
+
+    event.notify(ofs_event.ObjectWillBeMovedEvent(ob, self, id, self, new_id))
```

Comment:
@pbauer Yes indeed it's much faster but...
 In my case migration fails in many places with message like
`ERROR contentmigration Couldn't access */folder_MIGRATION_`
and then getObject() for catalog brain raises an error
`AttributeError: folder_MIGRATION_`
Still not sure why it happens. Looking in it...

Reply:
I am seeing similar problems where the catalog seems to end up unclean, having brains with paths like `/Plone/migrated_folder_MIGRATION_/item` which don't exists. It may depend on which order I do (first folders then items or the other way around), or which extra options I pass to the default and custom migrators (like reindexing or patching SearchableText). Sample error in the log is then:

`uncatalogObject unsuccessfully attempted to uncatalog an object with a uid of ...._MIGRATION_`

But it looks like this may be due to indexing queue optimisations in CMFCore (previously from `collective.indexing`). I don't see these errors when I start the instance with `CATALOG_OPTIMIZATION_DISABLED=1` in the environment and then do the migration.